internal/host/validator.go

```diff
@@ -372,6 +398,30 @@ func (v *validator) diskEncryptionRequirementsSatisfied(c *validationContext) Va
 		return ValidationPending
 	}
 
+	if hostutil.IsDay2Host(c.host) {
+		luks, err := v.getDiskEncryptionForDay2(c.host)
+		if err != nil {
+			return ValidationPending
+		}
+		if luks == nil {
+			// Disk encryption is disabled for workers on day1 cluster
+			return ValidationSuccessSuppressOutput
+		}
+
+		c.cluster.DiskEncryption = &models.DiskEncryption{}
+		if swag.BoolValue(luks.Clevis.Tpm2) {
+			// If Tpm2 is enabled for workers, check whether supported by the host.
+			c.cluster.DiskEncryption.Mode = swag.String(models.DiskEncryptionModeTpmv2)
+			return boolValue(c.inventory.TpmVersion == models.InventoryTpmVersionNr20)
+		} else if len(luks.Clevis.Tang) != 0 {
+			c.cluster.DiskEncryption.Mode = swag.String(models.DiskEncryptionModeTang)
+			return ValidationSuccess
+		} else {
+			// Only Tpm2 and Tang are available for disk encryption
+			return ValidationSuccessSuppressOutput
```

Comment:
That will mean that we have some malformed ignition, don't we want to fail the validation in this case?

Reply:
hmm, not sure it's necessarily means malformed. I mean, in case a new encryption type would be added, it might be a valid option later. Maybe I'll add a warning in this case then, to make it clearer.

Reply:
I don't think there will be one soon, and even if it will, ATM this can be caused only as an error in the ignition.
IMO, we better have a validation error here - in addition to what I said before, it will also ease the task of finding bugs if we reach this part of the code.

Reply:
ok will return ValidationFailure then. we could just remove it later on if a new mode is added.